libs/remix-ui/editor/src/lib/remix-ui-editor.tsx

```diff
@@ -541,6 +543,32 @@ export const EditorUI = (props: EditorUIProps) => {
       }
     })
 
+    editor.onDidPaste((e) => {
+       if (!pasteCodeRef.current && e && e.range && e.range.startLineNumber >= 0 && e.range.endLineNumber >= 0 && e.range.endLineNumber - e.range.startLineNumber > 10) {
+        const modalContent: AlertModal = {
+          id: 'newCodePasted',
+          title: 'New pasted code detected!',
+          message: (
+            <div> <i className="fas fa-exclamation-triangle text-danger mr-1"></i>
+              Remix detected you have just pasted a consequent code snippet or contract in the editor.
+              <div>
+                Please make sure you fully understand this new code before executing <b>any</b> transaction that uses it. 
```

Comment:
Make sure you fully understand this new code before executing <b>any</b> transaction that uses it.